cgi-bin/DW/Controller/MassPrivacy.pm

```diff
@@ -0,0 +1,159 @@
+      my $u = $rv->{u};
+      my $remote = $rv->{remote};
+
+    return "<?needlogin?>" unless $remote;
+    my ($s_dt, $e_dt, $posts);
+
+
+    return "This feature is currently disabled."
+        unless LJ::is_enabled('mass_privacy');
+
+    return error_ml('editprivacy.tt.unable') unless $u->can_use_mass_privacy;
+
+    my $mode = $POST->{'mode'} || $GET->{'mode'} || "init";
+    my $more_public = 0; # flag indiciating if security is becoming more public
+    # Check fields
+    my $errors = DW::FormErrors->new;
+    if ($r->did_post) {
+        return error_ml( 'error.invalidform' )  unless LJ::check_form_auth($POST->{lj_form_auth});
+
+        # Timeframe
+        $errors->add( "time", ".error.time" ) unless ($POST->{'time'});
+
+        # date range
+        if ($POST->{'time'} eq 'range' && $mode eq 'change') {
+            if ( !($POST->{'s_year'} =~ /\d+/) || !($POST->{'s_mon'} =~ /\d+/) || !($POST->{'s_day'} =~ /\d+/) ) {
+                $errors->add( "time", ".error.time.start" );
+            }
+            if ( !($POST->{'e_year'} =~ /\d+/) || !($POST->{'e_mon'} =~ /\d+/) || !($POST->{'e_day'} =~ /\d+/) ) {
+                $errors->add( "time", ".error.time.end" );
+            }
+
+            # Round down the day of month to the last day of the month
+            if ( $POST->{'s_day'} > LJ::days_in_month($POST->{'s_mon'}, $POST->{'s_year'}) ) {
+                $POST->{'s_day'} = LJ::days_in_month($POST->{'s_mon'}, $POST->{'s_year'});
+            }
+            if ( $POST->{'e_day'} > LJ::days_in_month($POST->{'e_mon'}, $POST->{'e_year'}) ) {
+                $POST->{'e_day'} = LJ::days_in_month($POST->{'e_mon'}, $POST->{'e_year'});
+            }
+        }
+
+        # security must change
+        if ($POST->{'s_security'} eq $POST->{'e_security'}) {
+            $errors->add( undef, ".error.security" );
+        }
+
+        # display initial page if errors
+        $mode = 'init' if $errors->exist;
+
+        # check if security is becoming more public
+        $more_public = 1 if ($POST->{'s_security'} eq 'private');
+        $more_public = 1 if ($POST->{'s_security'} eq 'friends' && $POST->{'e_security'} eq 'public');
+
+        if (($mode eq 'amsure') && $more_public && !LJ::auth_okay($u, $POST->{password}, undef, undef, undef)) {
+            $errors->add( undef, ".error.password" );
+            $mode = 'change' if $errors->exist;
+        }
+    }
+
+    # map security form values to 0) DB value 1) From string 2) To string
+    my %security = ( 'public'  => ['public', BML::ml('label.security.public2')],
+                     'friends' => ['usemask', BML::ml('label.security.accesslist')],
+                     'private' => ['private', BML::ml('label.security.private2')]);
+
+    my @security = ('public', BML::ml('label.security.public2'), 'friends', BML::ml('label.security.accesslist'), 'private', BML::ml('label.security.private2'));
+    # Initial view of page
+    if ($mode eq "init") {
```

Comment:
Why do we even have this if block? It seems like it doesn't do anything since there's no unqualified else?